core/src/test/kotlin/io/specmatic/core/CalculatePathTest.kt

```diff
@@ -726,4 +726,205 @@ internal class CalculatePathTest {
         
         assertThat(paths).containsExactlyInAnyOrder("{Person}.officeAddress{AddressRef}", "{Person}.homeAddress{Address}")
     }
+    
+    @Test
+    fun `calculatePath should handle deeply nested AnyPattern with typeAlias`() {
+        // Test case 1: Nested structure with typeAlias at multiple levels
+        val level3Pattern = JSONObjectPattern(
+            pattern = mapOf(
+                "data" to AnyPattern(listOf(StringPattern(), NumberPattern()))
+            ),
+            typeAlias = "Level3Object"
```

Comment:
Type aliases will typically be surrounded by round brackets. So they should be removed. Make sure the tests are updated such that all typeAliases are surrounded with round brackets, and update the code such that they are removed before being used (using the method `withoutPatternDelimiters` found in `core/io/specmatic/core/pattern/Grammar.kt`